cmd/server/controllers/section/section.go

```diff
@@ -0,0 +1,160 @@
+package section
+
+import (
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/natpapa17/MercadoFresco-ASociedadeGo/internal/sections"
+)
+
+type SectionController struct {
+	service sections.Service
+}
+
+func NewSection(s sections.Service) *SectionController {
+	return &SectionController{
+		service: s,
+	}
+}
+
+func (c SectionController) GetAll() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		s, err := c.service.GetAll()
+		if err != nil {
+			ctx.JSON(http.StatusNotFound, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+
+		ctx.JSON(http.StatusOK, gin.H{"data": s})
+	}
+}
+
+func (c SectionController) GetById() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		id, err := strconv.Atoi(ctx.Param("id"))
+		if err != nil {
+			ctx.JSON(http.StatusNotFound, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+
+		s, err := c.service.GetById(id)
+		if err != nil {
+			ctx.JSON(http.StatusNotFound, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+
+		ctx.JSON(http.StatusOK, s)
+	}
+}
+
+func (c *SectionController) Add() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+
+		body := ctx.Request.Body
+		data, _ := ioutil.ReadAll(body)
+
+		var obj sections.Section
+
+		sect := reflect.TypeOf(obj)
+		for i := 0; i < sect.NumField(); i++ {
+			field := sect.Field(i)
+			if !strings.Contains(string(data), field.Tag.Get("json")) && field.Tag.Get("json") != "id" {
+				ctx.JSON(http.StatusUnprocessableEntity, gin.H{"error": "missing field"})
+				return
+			}
+		}
```

Comment:
talvez fosse interessante remover esse 'for' e utilizar o shouldBindJson combinado com um tipo representando o body da request